apps/test/unit/applab/MockFirebaseTest.js

```diff
@@ -0,0 +1,93 @@
+import { expect } from '../../util/configuredChai';
+import MockFirebase from '../../util/MockFirebase';
+import { getDatabase } from '../../../src/applab/firebaseUtils';
+
+describe('MockFirebase', () => {
+  beforeEach(() => {
+    window.Applab = {
+      channelId: "test-firebase-channel-id",
+      firebaseName: 'test-firebase-name',
+      firebaseAuthToken: 'test-firebase-auth-token',
+    };
+  });
+
+  describe('once', () => {
+    it('calls callbacks', done => {
+      const firebase = new MockFirebase('https://example.firebaseio.com/');
+      firebase.set('foo');
+      firebase.once('value', snapshot => {
+          expect(snapshot.val()).to.equal('foo');
+          done();
+        },
+        error => console.warn(error));
+      firebase.flush();
+    });
+
+    it('resolves promises', done => {
+      const firebase = new MockFirebase('https://example.firebaseio.com/');
+      firebase.set('bar');
+      firebase.once('value').then(snapshot => {
+          expect(snapshot.val()).to.equal('bar');
+          done();
+        },
+        error => console.warn(error));
+      firebase.flush();
+    });
+  });
+
+  describe('set', () => {
+    it('calls callbacks', done => {
+      const firebase = new MockFirebase('https://example.firebaseio.com/');
+      firebase.set('foo', () => done());
+      firebase.flush();
```

Comment:
Interesting. I tried changing the 'resolves promises' tests to look like this:
```
     it('resolves promises', done => {
        const callback = sinon.spy();
        firebase.set('bar');
        firebase.once('value').then(
          snapshot => {
            expect(snapshot.val()).to.equal('bar');
            callback();
          },
          error => {throw error;});
        expect(callback.called).to.equal(false);
        firebase.flush();
        expect(callback.called).to.equal(true);
        done();
      });
 ```
However, they fail because the promises are not executed synchronously when `flush` is called, so `callback.called` is not `true` when we `expect` it to be. OTOH, 'calls callbacks' tests pass with this pattern. Weird, right? It's possible the delay is being introduced via the `new Promise(executor)` implementation, since that's the only added difference, but that's just a guess. 

Because using `flush` manually is a pain when multiple rounds of firebase calls are made, I am expecting most users of MockFirebase to use `autoFlush`, so I'm not going to spend any more time debugging this. 

I've changed all the tests here to use `autoFlush` instead of `flush`.